frontend/src/components/DirectMessage/Messaging.js

```diff
@@ -31,20 +32,18 @@ const Message = () => {
   };
 
   useEffect(() => {
-    // Effect hook for fetching users' information from Firestore
-    const unsubscribe = onSnapshot(
-      collection(db, "users_information"),
-      (snapshot) => {
-        const usersArray = [];
-        snapshot.forEach((doc) => {
-          usersArray.push({ id: doc.id, ...doc.data() });
-        });
-        setUsers(usersArray);
-      }
-    );
-    return unsubscribe; // Unsubscribe from the snapshot listener when the component unmounts
-  }, []);
-
+    const unsubscribe = onSnapshot(collection(db, 'users_information'), (snapshot) => {
+      const usersArray = [];
+      snapshot.forEach((doc) => {
+        usersArray.push({ id: doc.id, ...doc.data() });
+      });
+      setUsers(usersArray);
+    });
+    return () => {
+      unsubscribe();
+    };
+  }, [setUsers]);
```

Comment:
DO NOT MERGE, check if the setUsers is going to cause network issues as we had this issue before.